match/lib/match/importer.rb

```diff
@@ -2,23 +2,16 @@
 require_relative 'encryption'
 require_relative 'storage'
 require_relative 'module'
+require 'fastlane_core/provisioning_profile'
 require 'fileutils'
 
 module Match
   class Importer
-    def import_cert(params, cert_path: nil, p12_path: nil)
-      # Get and verify cert and p12 path
-      cert_path ||= UI.input("Certificate (.cer) path:")
-      p12_path ||= UI.input("Private key (.p12) path:")
-
-      cert_path = File.absolute_path(cert_path)
-      p12_path = File.absolute_path(p12_path)
-
-      UI.user_error!("Certificate does not exist at path: #{cert_path}") unless File.exist?(cert_path)
-      UI.user_error!("Private key does not exist at path: #{p12_path}") unless File.exist?(p12_path)
-
-      # Base64 encode contents to find match from API to find a cert ID
-      cert_contents_base_64 = Base64.strict_encode64(File.open(cert_path).read)
+    def import_cert(params, cert_path: nil, p12_path: nil, profile_path: nil)
+      # Get and verify cert, p12 and profiles path
+      cert_path = ensure_valid_file_path(cert_path, "Certificate", ".cer")
+      p12_path = ensure_valid_file_path(p12_path, "Private key", ".p12")
+      profile_path = ensure_valid_file_path(profile_path, "Provisioning profile", ".mobileprovision", optional: true)
```

Comment:
I think we need to also allow `.provisionprofile` here too for Mac profiles 🙃 Are we able to allow multiple types of extensions?